src/sleep/sleep.c

```diff
@@ -224,20 +224,23 @@ static int rtc_write_wake_alarm(uint64_t sec) {
         return 0;
 }
 
-static int execute_s2h(usec_t hibernate_delay_sec) {
-
-        _cleanup_strv_free_ char **hibernate_modes = NULL, **hibernate_states = NULL,
-                                 **suspend_modes = NULL, **suspend_states = NULL;
+static int execute_s2h(SleepConfig *sleep_config) {
+        char **suspend_modes       = NULL,
+             **suspend_states      = NULL,
+             **hibernate_modes     = NULL,
+             **hibernate_states    = NULL;
```

Comment:
I don't understand why those variables are needed. Why not just use `sleep_config->suspend_modes`?